partitura/musicanalysis/performance_codec.py

```diff
@@ -414,6 +462,101 @@ def tempo_by_average(
+        sampled (by default is None, which means that the tempo curve
+        is returned for each unique score onset)
+    return_onset_idxs : bool
+        Return the indices of the unique score onsets (Default is False)
+
+    Returns
+    -------
+    tempo_curve : np.ndarray
+        Tempo curve in seconds per beat (spb). If `input_onsets` was provided,
+        this array contains the value of the tempo in spb for each onset
+        in `input_onsets`. Otherwise, this array contains the value of the
+        tempo in spb for each unique score onset.
+    input_onsets : np.ndarray
+        The score onsets corresponding to each value of the tempo curve.
+    unique_onset_idxs: list
+        Each element of the list is an array of the indices of the score
+        corresponding to the elements in `tempo_curve`. Only returned if
+        `return_onset_idxs` is True.
+    """
+    # use float64, float32 led to problems that x == x + eps evaluated
+    # to True
+    score_onsets = np.array(score_onsets).astype(np.float64, copy=False)
+    performed_onsets = np.array(performed_onsets).astype(np.float64, copy=False)
+    score_durations = np.array(score_durations).astype(np.float64, copy=False)
+
+    performed_durations = np.array(performed_durations).astype(np.float64, copy=False)
+
+    # Get unique onsets if no provided
+    if unique_onset_idxs is None:
+        # Get indices of the unique onsets (quantize score onsets)
+        unique_onset_idxs = get_unique_onset_idxs((1e4 * score_onsets).astype(np.int))
+
+    # Get score information
+    score_info = get_unique_seq(onsets=score_onsets,
+                                offsets=score_onsets + score_durations,
+                                unique_onset_idxs=unique_onset_idxs,
+                                return_diff=False)
+    # Get performance information
+    perf_info = get_unique_seq(onsets=performed_onsets,
+                               offsets=performed_onsets + performed_durations,
+                               unique_onset_idxs=unique_onset_idxs,
+                               return_diff=False)
+
+    # unique score onsets
+    unique_s_onsets = score_info['u_onset']
+    # equivalent onsets
+    eq_onsets = perf_info['u_onset']
+
+    # Monotonize times
+    eq_onset_mt, unique_s_onsets_mt = monotonize_times(eq_onsets,
+                                                       deltas=unique_s_onsets)
+    # Function that that interpolates the equivalent performed onsets
+    # as a function of the score onset.
+    onset_fun = interp1d(unique_s_onsets_mt, eq_onset_mt, kind='linear',
+                         fill_value='extrapolate')
+
+    if input_onsets is None:
+        input_onsets = unique_s_onsets[:-1]
+
+    tempo_curve = derivative(onset_fun, input_onsets, dx=0.5)
+
+    output = [tempo_curve, input_onsets]
+
+    if return_onset_idxs:
+        output.append(unique_onset_idxs)
+
+    return output
```

Comment:
Return tuple instead of list of variable size

Reply:
sure